src/main/java/io/supertokens/multitenancy/Multitenancy.java

```diff
@@ -147,6 +147,51 @@ private static void validateTenantConfig(Main main, TenantConfig targetTenantCon
         // Verify that the keys in the coreConfig is valid
         validateConfigJsonForInvalidKeys(main, targetTenantConfig.coreConfig);
 
+        // Validate firstFactors and requiredSecondaryFactors
+        {
+            Set<String> disallowedFactors = new HashSet<>();
+            Map<String, String> factorIdToRecipeName = new HashMap<>();
+            if (!targetTenantConfig.emailPasswordConfig.enabled) {
+                disallowedFactors.add("emailpassword");
+
+                factorIdToRecipeName.put("emailpassword", "emailPassword");
+            }
+            if (!targetTenantConfig.passwordlessConfig.enabled) {
+                disallowedFactors.add("otp-email");
+                disallowedFactors.add("otp-phone");
+                disallowedFactors.add("link-email");
+                disallowedFactors.add("link-phone");
+
+                factorIdToRecipeName.put("otp-email", "passwordless");
+                factorIdToRecipeName.put("otp-phone", "passwordless");
+                factorIdToRecipeName.put("link-email", "passwordless");
+                factorIdToRecipeName.put("link-phone", "passwordless");
+            }
+            if (!targetTenantConfig.thirdPartyConfig.enabled) {
+                disallowedFactors.add("thirdparty");
+
+                factorIdToRecipeName.put("thirdparty", "thirdParty");
+            }
+
+            if (targetTenantConfig.firstFactors != null) {
+                for (String factor : targetTenantConfig.firstFactors) {
+                    if (disallowedFactors.contains(factor)) {
+                        throw new InvalidConfigException("firstFactors should not contain '" + factor
```

Comment:
In order to have factor in firsFactors, emailPasswordEnabled should be true in the tenant config.